upstairs/src/upstairs.rs

```diff
@@ -2008,17 +2008,8 @@ impl Upstairs {
         self.downstairs
             .notify_nexus_of_client_task_stopped(client_id, reason);
 
-        // If the upstairs is already active (or trying to go active), then the
-        // downstairs should automatically call PromoteToActive when it reaches
-        // the relevant state.
-        let auto_promote = match self.state {
```

Comment:
`auto_promote` is a deterministic function of `self.state`, so we don't need to compute it here (because we're passing `&self.state` into the function below).